common/oauthTokenManager.go

```diff
@@ -713,15 +718,85 @@ func (credInfo *OAuthTokenInfo) GetNewTokenFromMSI(ctx context.Context) (*adal.T
 	if credInfo.IdentityInfo.MSIResID != "" {
 		params.Set("msi_res_id", credInfo.IdentityInfo.MSIResID)
 	}
+
 	req.URL.RawQuery = params.Encode()
 	req.Header.Set("Metadata", "true")
+
 	// Set context.
 	req.WithContext(ctx)
 
 	// Send request
 	resp, err := msiTokenHTTPClient.Do(req)
+
+	return req, resp, err
+}
+
+// fixupTokenJson corrects the value of JSON field "not_before" in the Byte slice from blank to a valid value and returns the corrected Byte slice.
+
+// Dated 15th Sep 2021.
+// Token JSON returned by ARC-server endpoint API currently does not set a valid integral value for "not_before" key.
+// If the token JSON already has "not_before" correctly set, this will be a no-op.
+func fixupTokenJson(bytes []byte) []byte {
+	byteSliceToString := string(bytes)
+	separatorString := `"not_before":"`
+	stringSlice := strings.Split(byteSliceToString, separatorString)
+
+	if stringSlice[1][0] != '"' {
```

Comment:
Minor: is "not_before" always part of the json? Perhaps it's best to check for it before invoking stringSlice[1] just to make sure.

Reply:
Yes, it is always a part of the JSON - Be it a response from the Arc IMDS endpoint (with an empty string value), or Azure VM (with a non-empty string value)